core/src/main/java/org/apache/iceberg/BaseTable.java

```diff
@@ -63,6 +63,10 @@ public String name() {
     return name;
   }
 
+  public MetricsReporter reporter() {
```

Comment:
I don't think we should expose this. We should rather test that `report()` of the custom reporter is actually being called

Reply:
sure.